OpenSim/Simulation/Manager/Manager.cpp

```diff
@@ -789,13 +804,21 @@ void Manager::initialize(const SimTK::State& s)
  * 
  * @param s system state before integration
  */
-void Manager::finalize(SimTK::State& s )
+void Manager::finalize(const SimTK::State& s )
 {
         // ANALYSES 
     if(  _performAnalyses ) { 
         AnalysisSet& analysisSet = _model->updAnalysisSet();
         analysisSet.end(s);
     }
+    if (_writeToStorage) {
+        SimTK::Vector stateValues = _model->getStateVariableValues(s);
+        StateVector vec;
+        vec.setStates(s.getTime(), stateValues);
+        getStateStorage().append(vec);
+        if (_model->isControlled())
+            _controllerSet->storeControls(s, getStateStorage().getSize());
+    }
```

Comment:
This duplication is not ideal. Perhaps at least place a comment saying that this is duplicated, so that maintainers don't accidentally only update one location?

Reply:
I agree. Eliminated that duplication.